### DISCUSSION TITLE
What is the difference from the VS Code official by server-main.js or server-cli.js startup?

### DISCUSSION THREAD

**Discussion by yiliang114** [coder/code-server#6222] [Q&A] [answered]
Through the source code, I learned that code-server(this project) use `httpServer` monitoring access and ran the Node part of VS Code in the form of sub-processes, and did a lot of other work, such as login verification, cross-platform packaging, etc.

However, I also have a certain understanding of the official code of VS Code. It seems that such remote mode can be run directly through the official entry file `server-main.js` or `server-cli.js`.  

Both startup methods seem to be the same, is there any reason for code-server to implement it (about startup method on server) all over again?

**Comment by code-asher** [comment, marked as answer]
You could run upstream's server directly if you want.  We make these changes to it: https://github.com/coder/code-server/blob/main/docs/FAQ.md#whats-the-difference-between-code-server-and-github-codespaces

Some of those changes are patches and some of them are in the wrapper.  Part of the reason we did it this way instead of patching everything into Code is that maintaining patches is more difficult so we prefer to patch as little as possible and have the rest in the wrapper.

**Reply by yiliang114** [reply]
@code-asher I found a difference, the official code-server.sh of vscode is just a development debug mode, meaning that the static resources it loads are not bundled. This leads to the fact that if I create a docker image directly and use code-server.sh as the startup command, the page load is very slow.

The reason for this is that  coder/code-server are loading a reh and min version of the resource. 

It seems that vscode does not provide out-of-the-box docker solutions.

**Reply by code-asher** [reply]
Yeah I think you would need to build it, or install VS Code and run `code serve-web` which seems to download a minified server for you.
